src/Uno.UI.Composition/Composition/CompositionVisualSurface.skia.cs

```diff
@@ -36,9 +37,19 @@ void ISkiaSurface.UpdateSurface(bool recreateSurface)
 					}
 				};
 
+				if (_displayInformation is null)
```

Comment:
We need to find a different way to do this, as this would not work in multi-window scenarios. I assume the correct approach will be to use `RasterizationScale`, and listening to `XamlRoot.Changed` for changes

Reply:
Is there a way to get access to `XamlRoot` from the Composition side?

Reply:
I think that is not the case, in multi-window WinUI app, each Window has its own `RasterizationScale`, so when you have multiple screens and move one window to another screen, it will adjust to the scaling setting on that specific monitor

Reply:
I see, any idea on how to get the `XamlRoot` to fix this? I don't fully remember but there is a separate CompositionTarget per window, right? maybe we can save an XamlRoot reference there?

Reply:
Yes, I think that is the case, we should now have separate CompositionTargets. So we should be able to keep the XamlRoot there (although not 100% correct I guess, but should be good enough for now)

Reply:
fixed